src/DataBus/ConfigureAzureDataBus.cs

```diff
@@ -162,31 +145,14 @@ public static DataBusExtensions<AzureDataBus> CleanupInterval(this DataBusExtens
         }
 
         /// <summary>
-        /// Sets token credential to authenticate with Storage Blob service
-        /// <remarks>Token credentials can be created using <see cref="AzureServiceTokenProvider"/> with token renewal configured.</remarks>
+        /// For advanced authentication scenarios, provide a properly configured BlobContainerClient instead of a connectionstring
         /// </summary>
-        /// <param name="config">The configuration to modify.</param>
-        /// <param name="storageAccountName">The storage account name.</param>
-        /// <param name="renewalTimeBeforeTokenExpires">The renewal time before token expiry.</param>
-        /// <param name="endpointSuffix">Endpoint suffix associated with storage blob service per type of data center. Default is "core.windows.net" for public Azure data centers.</param>
-        public static DataBusExtensions<AzureDataBus> AuthenticateWithManagedIdentity(this DataBusExtensions<AzureDataBus> config, string storageAccountName, TimeSpan renewalTimeBeforeTokenExpires, string endpointSuffix = "core.windows.net")
+        public static DataBusExtensions<AzureDataBus> BlobContainerClient(this DataBusExtensions<AzureDataBus> config,
+            BlobContainerClient blobContainerClient)
         {
-            if (renewalTimeBeforeTokenExpires <= TimeSpan.Zero)
-            {
-                throw new ArgumentException($"Should not be less or equal to {nameof(TimeSpan.Zero)}", nameof(renewalTimeBeforeTokenExpires));
-            }
-
-            if (string.IsNullOrWhiteSpace(storageAccountName))
-            {
-                throw new ArgumentException("Should not be null or empty", nameof(storageAccountName));
-            }
-
-            var dataBusSettings = GetSettings(config);
-
-            dataBusSettings.RenewalTimeBeforeTokenExpires = renewalTimeBeforeTokenExpires;
-            dataBusSettings.StorageAccountName = storageAccountName;
-            dataBusSettings.EndpointSuffix = endpointSuffix;
-
+            Guard.AgainstNull(nameof(blobContainerClient), blobContainerClient);
+            
+            config.GetSettings().Set(blobContainerClient);
```

Comment:
Should this use the [settings key](https://github.com/Particular/NServiceBus.DataBus.AzureBlobStorage/pull/138/files#diff-e9e485642dfe749053a525eb48e17c703acec7f52db5646e39f3d9bb3b01ffebR3)?